sky/execution.py

```diff
@@ -378,55 +379,120 @@ def spot_launch(
               f'{colorama.Style.RESET_ALL}')
         return
 
+    # ================================================================
```

Comment:
Is it possible / easy to refactor L382 -- end of translation to a helper? If not, no worries.